sqlmesh/core/engine_adapter/clickhouse.py

```diff
@@ -296,6 +299,64 @@ def _build_partitioned_by_exp(
             this=exp.Schema(expressions=partitioned_by),
         )
 
+    def inject_query_setting(
```

Comment:
This method seems too generic, and just by glancing at it, it's impossible to tell when I should be calling this vs not. Nowhere in the name / signature of this method there's any indication that it's only for SCD2 / table diffing / other cases that involve joins and rely on null values.

I understand the motivation to make it more abstract, but in this case it seems to be causing more confusion than clarity.

Can we perhaps make it purpose-driven for now, until a better abstraction emerges (if it ever does). Eg. something like:
```
def ensure_nulls_after_join
```
or something similar that would indicate the purpose and the outcome.

Reply:
yep - that was my initial inclination, but i wasn't sure if it leaked too much CH-specific stuff into the base adapter